torch/_subclasses/meta_utils.py

```diff
@@ -302,6 +304,30 @@ def empty_create(inner_t, inner_src, symbolic_context=symbolic_context):
                         with torch.enable_grad():
                             r = r.clone()
                             r._coalesced_(t.is_coalesced())
+                    # The meta tensor does not carry all sparsity related meta
+                    # data (yet), so we just store sparse tensor meta data
+                    # directly in the FakeTensor for now.
+                    r.sparsity = _extract_sparse_tensor_metadata(t)
+                elif t.layout is not torch.strided:
+                    # Handle sparse tensors (other than COO handled above, which
+                    # has to account for colaescing). Passing in t.to('meta')
+                    # as lambda does not work (yet). Therefore, we pass in
+                    # a dummy meta tensor.
+                    is_leaf = safe_is_leaf(t)
+                    r = callback(
+                        # TODO: replace with proper compressed meta tensor
+                        lambda: torch.tensor(t.shape, dtype=t.dtype, device="meta")
```

Comment:
This is a blocker as meta device is not supported in sparse compressed tensor constructors. I'll try to cook a PR for this..

Reply:
Yeah, it is possible that by providing the meta device support, a lot of the code I added will "just work"

Reply:
https://github.com/pytorch/pytorch/pull/120588 adds fake tensor support to sparse compressed tensors. When it lands, modification of meta_utils.py in this PR ought to be unnecessary.

Reply:
Wait, no changes to meta_utils.py at all? We should still have the coo and csr/csc/bsr/bsc cases that handle setting up the "meta" data, right? After that most code indeed goes away all the way down to setting up (sparse) tensor metdata?

Reply:
Ah, I see. With #120588 includes a similar branching structure. Okay, I will wait until all those are merged and then see what is left to do ;-). Thanks for helping out with all this.